File: javascript/difference-of-squares/difference-of-squares.js
function Squares(limit) {
	this.squareOfSums = this._square_of_sums(limit);
	this.sumOfSquares = this._sum_of_squares(limit);
	this.difference = this.squareOfSums - this.sumOfSquares;
}

Squares.prototype._square_of_sums = function(limit) {
	var sum = 0;
	for (var i = 2; i <= limit; ++i) {
		sum += i;
	}
	
	return Math.pow(sum + 1, 2);
};

Squares.prototype._sum_of_squares = function(limit) {
	var sum = 0;
	for (var i = 2; i <= limit; ++i) {
		sum += Math.pow(i, 2);
	}
	
	return sum + 1;
};

module.exports = Squares;

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(limit):
	sq_sum = 0
	limit += 1
	for i in range(1, limit):
		sq_sum += i

	return sq_sum ** 2

def sum_of_squares(limit):
	sum_sq = 0
	limit += 1
	for i in range(1, limit):
		sum_sq += i ** 2
	
	return sum_sq


def difference(limit):
	sq_sum = square_of_sum(limit)
	sum_sq = sum_of_squares(limit)
	
	return sq_sum - sum_sq